src/super_gradients/training/metrics/depth_estimation_metrics.py

```diff
@@ -0,0 +1,250 @@
+from typing import Tuple, Sequence, Union, Optional
+
+import torch
+from torch import Tensor
+from torchmetrics import MeanSquaredError, MeanSquaredLogError, MeanAbsoluteError, MeanAbsolutePercentageError, Metric
+from super_gradients.common.object_names import Metrics
+from super_gradients.common.registry import register_metric
+
+
+class DepthEstimationMetricMixin:
+    """
+    A mixin class providing common functionality for depth estimation metrics.
+
+    :param ignore_val: Value to be ignored when computing metrics.
+    :param apply_sigmoid: Whether to apply the sigmoid function to predictions before updating the metric.
+    """
+
+    def __init__(self, ignore_val: Optional[float] = None, apply_sigmoid: bool = False):
+        self.ignore_val = ignore_val
+        self.apply_sigmoid = apply_sigmoid
+
+    def process_preds_and_target(self, preds: Union[Tensor, Sequence[Tensor]], target: Tensor) -> Tuple[Tensor, Tensor]:
```

Comment:
Do we need this `DepthEstimationMetricMixin ` class only as a "utils toolkit" ?

If so, maybe we could simplify the code/inheritance by replacing this mixin by a function:

```python
def format_depth_preds_and_targets(preds: Union[Tensor, Sequence[Tensor]], target: Tensor, ignore_val: Optional[float] = None, apply_sigmoid: bool = False)
```

And then use it like this


```python
class MeanSquaredErrorWithIgnored(MeanSquaredError):
    def __init__(self, squared: bool = True, ignore_val: Optional[float] = None, apply_sigmoid: bool = False):
        super().__init__(squared=squared)
        self.ignore_val = ignore_val
        self.apply_sigmoid = apply_sigmoid

    def update(self, preds: Tensor, target: Tensor) -> None:
        preds, target = format_depth_preds_and_targets(preds, target, ignore_val=self.ignore_val, apply_sigmoid=self.apply_sigmoid)
        super().update(preds, target)
```


Alternatively, we can always have keep the class, but use it with composition instead of inheritence

```python
class MeanSquaredErrorWithIgnored(MeanSquaredError):
    def __init__(self, squared: bool = True, ignore_val: Optional[float] = None, apply_sigmoid: bool = False):
        super().__init__(squared=squared)
        # Note: We should rename `DepthEstimationMetricMixin` in this case
        self.formater = DepthEstimationMetricMixin(ignore_val=ignore_val, apply_sigmoid=apply_sigmoid)

    def update(self, preds: Tensor, target: Tensor) -> None:
        preds, target =self.formater.process_preds_and_target(preds, target)
        super().update(preds, target)
```

IMO it makes the inheritance logic simpler and less confusing. It also emphasizes on the facts these are just internal utils/formating steps, and not really a "contract" that other classes need to know about.
What do you think?

Reply:
For me it is more clear this way, but I might be biased. I also think the code is cleaner and more consistent when inheriting from the mixin class.
@BloodAxe Whats your opinion ?

Reply:
1) A "canonical" mixin should not have a constructor. While the `DepthEstimationMetricMixin` has. So we are dealing with multiple inheritance class here. Not necessary a bad thing, but maybe the name of the base class should be changed and not use `mixin` word to avoid confusion.

2) Another thing that worries me is the naming of the depth metrics. I think they should have a common name prefix like "Depth"-bla-bla-bla. This is to discriminate depth-related metrics from other tasks.

3) A dedicated class just to support ignore value looks like a great overkill from my perspective `MeanSquaredErrorWithIgnored`. We should have one class (MeanSquaredError) that would allow user to set the `ignore_value` attribute. I just don't see any benefit in having second class here. I cann't see a use case where you want to compute both  `MeanSquaredError` and `MeanSquaredErrorWithIgnored` metrics simultaneously (Which can justify presence of two classes - to get metrics a unique names).

Reply:
Ok went with your suggestion which we discussed (similar idea to what you suggested @Louis-Dupont , but the other way around - the base class receives the metric instead).
Added depth prefixes too.